specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+the function MUST emit the `VoteCast` event. The function MUST also return the voting weight.
+
+#### `castVoteWithReasonAndParamsBySig`
+
+Cast a vote on a proposal with a reason and additional encoded parameters using a signature.
+
+```solidity
+function castVoteWithReasonAndParams(uint256 _proposalId, uint8 _support, string memory _reason, bytes memory _params, uint8 _v, bytes32 _r, bytes32 _s) external returns (uint256)
+```
+
+This function MUST emit the `VoteCastWithParams` event if the length of the parameters is more than zero. Otherwise,
+the function MUST emit the `VoteCast` event. The function MUST also return the voting weight.
+
+#### `editProposalType`
+
+Edit the type of a proposal that is still active. This function MUST only be callable by the manager.
+
+```solidity
+function editProposalType(uint256 _proposalId, uint8 _proposalType) external
+```
+
+This function MUST revert if the proposal type is not already configured. Additionally, this function MUST emit
+the `ProposalTypeUpdated` event.
+
+#### `execute`
+
+Execute a successful proposal. This MUST only be possible when the quorum is reached, the vote is successful, and the
+deadline is reached.
+
+```solidity
+function execute(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, bytes32 _descriptionHash) external payable returns (uint256)
+```
+
+This function MUST emit the `ProposalExecuted` event and return the proposal ID.
+
+#### `executeWithModule`
+
+Execute a successful proposal that was created with a module. This MUST only be possible when the quorum is reached,
+the vote is successful, and the deadline is reached.
+
+```solidity
+function executeWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash) external payable returns (uint256)
+```
+
+This function MUST emit the `ProposalExecuted` event and return the proposal ID.
+
+#### `propose`
+
+Creates a new proposal with a given proposal type. This function MUST check that the proposal type exists.
+
+```solidity
+function propose(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, string memory _description) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCreated` event and return the ID of the proposal created.
+
+#### `proposeWithModule`
+
+Creates a new proposal with a given proposal type and module. This function MUST check that the proposal type exists,
+and that the module is supported by the `Governor` contract.
+
+```solidity
+function proposeWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash, uint8 _proposalType) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCreated` event and return the ID of the proposal created.
+
+#### `relay`
+
+Relays a transaction or function call to an arbitrary target. This function MUST only be callable as part of a
```

Comment:
Interesting, why is this a public function?

Reply:
added a section to explain:

> In cases where the governance executor is some contract other than the `Governor` itself, like when using a timelock, this function CAN be invoked in a governance proposal to recover tokens or Ether that was sent to the `Governor` contract by mistake.